src/metabase/api/setup.clj

```diff
@@ -86,4 +88,108 @@
       (catch Throwable e
         (response-invalid :general (.getMessage e))))))
 
+
+;;; Admin Checklist
+
+(defn- admin-checklist-values []
+  (let [has-dbs?           (db/exists? Database)
+        has-dashboards?    (db/exists? 'Dashboard)
+        has-pulses?        (db/exists? 'Pulse)
+        has-labels?        (db/exists? 'Label)
+        has-hidden-tables? (db/exists? 'Table, :visibility_type [:not= nil])
+        has-metrics?       (db/exists? 'Metric)
+        has-segments?      (db/exists? 'Segment)
+        num-tables         (db/select-one-count 'Table)
+        num-cards          (db/select-one-count 'Card)
+        num-users          (db/select-one-count 'User)]
+    [{:title       "Add a database"
+      :group       "Get connected"
+      :description "TODO - Write something good here"
+      :link        "/admin/databases/create"
+      :completed   has-dbs?
+      :triggered   :always}
```

Comment:
I'll just note that I didn't recommend an approach - 
My nudges were towards-
* Testable logic
* Not breaking up logic between the front-end and back-end

I don't feel strongly on this however. We'll be adding steps and I'm willing to bet $20 that at in the next few additional steps we'll find something that requires us to refactor this in some way.

I predict that we will need to cache this somehow. That's the main thing I care about and want to test (on heroku or equally crappy hosts) and confirm.